source/Octopus.Tentacle.Client/Execution/RpcCallExecutor.cs

```diff
@@ -84,7 +84,7 @@ public async Task<T> ExecuteWithRetries<T>(
 
                             onErrorAction?.Invoke(lastException);
 
-                            var remainingDurationInSeconds = (int)(totalRetryDuration - elapsedDuration).TotalSeconds;
+                            var remainingDurationInSeconds = Math.Max((int)(totalRetryDuration - elapsedDuration).TotalSeconds, 0);
                             logger.Info($"An error occurred communicating with Tentacle. This action will be retried after {(int)sleepDuration.TotalSeconds} seconds. Retry attempt {retryCount}. Retries will be performed for up to {remainingDurationInSeconds} seconds.");
```

Comment:
This logging could be better, we probably should be taking into account the min number of retries.

I am opting not to do it, it would be ideal but the value it provides is to low given the time remaining on the project.